syz-verifier/main.go

```diff
@@ -298,6 +303,38 @@ func (vrf *Verifier) generate() (*prog.Prog, int) {
 	return vrf.target.Generate(vrf.rnd, prog.RecommendedCalls, vrf.choiceTable), vrf.progIdx
 }
 
+// processResults will send a set of complete results for verification and, in
+// case differences are found, it will store a result report highlighting those
+// in th workdir/results directory. If writing the results fails, it returns an
+// error.
+func (vrf *Verifier) processResults(res []*verf.Result, prog *prog.Prog) error {
+	rr := verf.Verify(res, prog)
+	if rr == nil {
+		log.Print("HEREEEEE")
+		return nil
+	}
+	data := fmt.Sprintf("Errno mismatches found for program:\n%s\n", rr.Prog)
+	data += "CALL REPORTS FOUND BELOW\n"
+
+	mismatchIdx := 0
+	for idx, cr := range rr.Reports {
+		data += fmt.Sprintf("Report for call: %s", cr.Call)
+
+		if mismatchIdx < len(rr.Mismatch) && idx == rr.Mismatch[mismatchIdx] {
+			data += " - MISMATCH FOUND"
+			mismatchIdx++
+		}
+		data += "\n"
+		for pool, errno := range cr.Errnos {
+			data += fmt.Sprintf("Pool: %d, Errno: %d\n", pool, errno)
+		}
+		data += "\n"
+	}
+	err := osutil.WriteFile(filepath.Join(vrf.resultsdir, fmt.Sprintf("result-%d", vrf.resIdx)), []byte(data))
```

Comment:
There are 2 problems with this:
1. If I left it running over a weekend, I will get back to a dead machine with overflowed disk (maybe even with no possibility to login, because that may want to write some files, which will be failing).
2. After restart it will start overwriting and losing old results.

Reply:
As a solution, should we connect and store results in some remote database or should I just stop writing results (and potentially stop the syz-verifier process) if the machine runs out of space. I presume running out of memory is indeed the drawback of saving reports locally.

Reply:
I would with how syz-manager stores info about crashes, it avoids both issues.